app/src/main/java/app/vdh/org/vdhapp/feature/report/presentation/map/viewmodel/ReportMapViewModel.kt

```diff
@@ -1,39 +1,47 @@
-package app.vdh.org.vdhapp.viewmodels
+package app.vdh.org.vdhapp.feature.report.presentation.map.viewmodel
 
 import android.app.Application
 import android.util.Log
 import androidx.lifecycle.AndroidViewModel
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.liveData
 import androidx.lifecycle.switchMap
 import androidx.lifecycle.viewModelScope
-import app.vdh.org.vdhapp.api.Result
-import app.vdh.org.vdhapp.consts.PrefConst
-import app.vdh.org.vdhapp.data.ReportRepository
-import app.vdh.org.vdhapp.data.SingleLiveEvent
-import app.vdh.org.vdhapp.data.entities.ReportEntity
-import app.vdh.org.vdhapp.data.actions.ReportMapAction
-import app.vdh.org.vdhapp.data.actions.ReportMapFilterViewAction
-import app.vdh.org.vdhapp.data.actions.ReportMapViewAction
-import app.vdh.org.vdhapp.data.models.BikePathNetwork
-import app.vdh.org.vdhapp.data.models.BoundingBoxQueryParameter
+import app.vdh.org.vdhapp.core.helpers.CallResult
+import app.vdh.org.vdhapp.core.consts.PrefConst
+import app.vdh.org.vdhapp.core.helpers.SingleLiveEvent
+import app.vdh.org.vdhapp.feature.report.domain.common.model.ReportModel
+import app.vdh.org.vdhapp.feature.report.domain.map.model.BikePathNetwork
+import app.vdh.org.vdhapp.feature.report.domain.map.model.BoundingBoxQueryParameter
+import app.vdh.org.vdhapp.feature.report.domain.map.usecase.GetBicyclePathUseCase
+import app.vdh.org.vdhapp.feature.report.domain.map.usecase.GetReportListUseCase
+import app.vdh.org.vdhapp.feature.report.presentation.map.action.ReportMapAction
+import app.vdh.org.vdhapp.feature.report.presentation.map.action.ReportMapFilterViewAction
+import app.vdh.org.vdhapp.feature.report.presentation.map.action.ReportMapViewAction
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
 
-class ReportMapViewModel(app: Application, private val repository: ReportRepository) : AndroidViewModel(app) {
+class ReportMapViewModel(
+    app: Application,
+    reportListUseCase: GetReportListUseCase,
+    private val bicyclePathUseCase: GetBicyclePathUseCase
+) : AndroidViewModel(app) {
```

Comment:
Do you really need this to be extending `AndroidViewModel` ?

Reply:
Not useful in this VM you're right